bytecomp/bytepackager.ml

```diff
@@ -278,10 +295,10 @@ let package_object_files ~ppf_dump files targetfile targetname coercion =
       { cu_name = targetname;
         cu_pos = pos_code;
         cu_codesize = pos_debug - pos_code;
-        cu_reloc = List.rev !relocs;
+        cu_reloc = List.rev state.relocs;
         cu_imports =
           (targetname, Some (Env.crc_of_unit targetname)) :: imports;
-        cu_primitives = !primitives;
+        cu_primitives = List.rev state.primitives;
```

Comment:
Why is this reversed?

Reply:
primitives are accumulated in reverse order, I just put them back in the order they were seen, similar to what's done for cu_reloc.